lib/clamav/commands/instream_command.rb

```diff
@@ -19,10 +19,11 @@
 module ClamAV
   module Commands
     class InstreamCommand < Command
+      DEFAULT_MAX_CHUNK_SIZE = 1024
 
-      def initialize(io, max_chunk_size = 1024)
+      def initialize(io)
```

Comment:
This constructor being part of the public API, I think that this could break some people's code.

Let's maybe change https://github.com/franckverrot/clamav-client/blob/master/lib/clamav/client.rb#L28 and allow for another parameter to be passed, like `configuration` (a global configuration object):

```ruby
class ClamAV::Configuration
  attr_reader :instream_max_chunk_size
  def initialize(instream_max_chunk_size: 1024) # we can for sure use a const here too
    @instream_max_chunk_size = instream_max_chunk_size
  end
end

class ClamAV::Client
  def initialize(connection = default_connection, configuration: Configuration.new)
  ...
  @configuration = configuration
...

    def instream(io)
      execute Commands::InstreamCommand.new(io, @configuration.instream_max_chunk_size)
    end

...
```

allowing us to do something like


```ruby
configuration = Configuration.new(instream_max_chunk_size: 2048)
client = Client.new(configuration: configuration)
...
```

and get to pass down the value this way?

Reply:
Let me update my PR